### PR TITLE
Main timeline styling and DirectorShot introduction

### PR DESCRIPTION
**Pull request by Markvanderuit**
This PR fixes an issue with the director timeline, and provides miscellaneous styling to the timelines, and some modal views.

### CODE REVIEW

**Comment by martijnjanssen** [comment]
I see a lot of failed builds, I guess this is caused by checkstyle?

**Comment by Markvanderuit** [comment]
Yes.

File: src/main/java/control/DirectorTimelineController.java [hunk 1]
```diff
@@ -1,16 +1,186 @@
...
+     */
+    private void checkCollisions(DirectorShotBlock directorShotBlock) {
+        DirectorTimeline timeline = controllerManager.getScriptingProject()
+                .getDirectorTimeline();
+
+        // Remove overlaps for non-colliding shotblocks
+        ArrayList<DirectorShotBlock> toRemove = new ArrayList<>();
+        this.overlappingShotBlocks.stream()
+                .filter(shotBlock ->
+                            shotBlock.getShot().getCollidesWith().isEmpty()).forEach(shotBlock -> {
```
Comment (mennoo1996) [inline src/main/java/control/DirectorTimelineController.java]: Can you put the forEach on a new line. It looks like a nested lambda now

Comment (Markvanderuit) [inline src/main/java/control/DirectorTimelineController.java]: Done.

File: src/main/java/gui/centerarea/CameraTimetableBlock.java [hunk 2]
```diff
@@ -35,18 +35,18 @@ void initNormalPane() {
 
     }
 
-    @Override
+    /**
+     * Initialize the dragpane with blur, tutle, count and description labels.
```
Comment (mennoo1996) [inline src/main/java/gui/centerarea/CameraTimetableBlock.java]: tutle?

**Comment by mennoo1996** [comment]
Good work, two small comments

**Comment by bart-de-jonge** [comment]
Maybe fix checkstyle along the way next time. These streaks of failing builds are generally unwanted behaviour

File: src/main/java/control/DirectorTimelineController.java [hunk 3]
```diff
@@ -1,16 +1,187 @@
...
+     * @param endCount End count
+     * @param frontPadding additional recording time at the front
+     * @param endPadding additional recording time at the end
+     * @param cameras the cameras to generate the shot to
+     */
+    public void addDirectorShot(String name, String description,
+                                double startCount, double endCount,
+                                double frontPadding, double endPadding,
+                                List<Integer> cameras) {
+        log.info("Adding DirectorShot to DirectorTimeline");
```
Comment (bart-de-jonge) [inline src/main/java/control/DirectorTimelineController.java:63]: When creating the DirectorTimelineController you also log the actual controllerManager, maybe also do that here with all parameters?

File: src/main/java/control/DirectorTimelineController.java [hunk 4]
```diff
@@ -1,16 +1,187 @@
...
+        controllerManager.setActiveShotBlock(shotBlock);
+        this.shotBlocks.add(shotBlock);
+        controllerManager.getScriptingProject().changed();
+
+        // Check for collisions
+        checkCollisions(shotBlock);
+    }
+
+    /**
+     * Handle updated director shot. The previous timeline is used to retrieve the corresponding
```
Comment (bart-de-jonge) [inline src/main/java/control/DirectorTimelineController.java:83]: There is no switching of timelines going on when drag-and-droping directorshots. So please remove the parts of the javadoc that states that

**Comment by bart-de-jonge** [comment]
For now it looks fine, added a few comments so please take a look at them. Currently timetableblock still has logic for distinguishing camera and director blocks/timelines. Please abstract this in some way (interface/abstract methods) in a future pr?

**Comment by Markvanderuit** [comment]
@martijn9612 Can you look at the abstraction bart mentioned? I have some homework I need to do first, so I can't look into it until later tonight.

**Pull request merged**